debian/res/okadminfinder.1
.TH OKADMINFINDER 1 "July 2024" "version 1.1.0" "okadminfinder Manual"
.SH NAME
okadminfinder \- Easy way to find [Admin panel - Directories - Subdomains] of a website
.SH SYNOPSIS
.B okadminfinder
[\fIOPTION\fR]...
.SH DESCRIPTION
OKadminFinder is a tool designed to identify and locate administrative panels, directories, and subdomains on a target website. It is useful for penetration testers, security researchers, and ethical hackers.
.SH OPTIONS
.TP
\fB\-h\fR, \fB\-\-help\fR
Show this help message and exit.
.TP
\fB\-u\fR \fBURL\fR, \fB\-\-url\fR \fBURL\fR
Specify the target URL (e.g. 'www.example.com' or 'example.com').
.TP
\fB\-d\fR, \fB\-\-dns\fR
Enable DNS subdomain finding.
.TP
\fB\-w\fR \fBfile\fR, \fB\-\-wordlist\fR \fBfile\fR
Use a custom wordlist (e.g. /usr/share/wordlist/sub-domains.txt).
.TP
\fB\-t\fR, \fB\-\-tor\fR
Use the Tor anonymity network to conceal your IP address.
.TP
\fB\-p\fR \fBPROXY\fR, \fB\-\-proxy\fR \fBPROXY\fR
Use an HTTP(S) proxy (e.g. '127.0.0.1:8080') or a SOCKS(4/5) proxy (e.g. 'socks5://127.0.0.1:8080').
.TP
\fB\-r\fR, \fB\-\-random\-agent\fR
Use a randomly selected User-Agent for each request.
.SH EXAMPLES
.TP
\fBokadminfinder \-u example.com\fR
Scan the target website example.com for admin panels, directories, and subdomains.
.TP
\fBokadminfinder \-u example.com \-d\fR
Scan the target website example.com for DNS subdomains.
.TP
\fBokadminfinder \-u example.com \-w /path/to/wordlist.txt\fR
Scan the target website example.com using a custom wordlist located at /path/to/wordlist.txt.
.TP
\fBokadminfinder \-u example.com \-t\fR
Scan the target website example.com using the Tor network.
.TP
\fBokadminfinder \-u example.com \-p 127.0.0.1:8080\fR
Scan the target website example.com using an HTTP(S) proxy at 127.0.0.1:8080.
.TP
\fBokadminfinder \-u example.com \-r\fR
Scan the target website example.com using a randomly selected User-Agent for each request.
.TP
\fBokadminfinder \-u example.com \-D\fR
Enable Debug mode.
.SH BUGS
Report bugs to https://github.com/michyamrane/okadminfinder/issues.
.SH AUTHOR
OKadminFinder was developed by mIcHy AmRaNe <6m63er17c@mozmail.com>.
.SH COPYRIGHT
This is free software: you are free to change and redistribute it. There is no warranty, to the extent permitted by law.


debian/changelog
okadminfinder (1.1.1-1) stable; urgency=low

  * Debug mode.
  * build the deb package using pyproject.toml instead of setup.py

 -- mIcHy AmRaNe <6m63er17c@mozmail.com>  Thu, 11 Jul 2024 17:29:05 +0200

okadminfinder (1.1.0-1) stable; urgency=low

  * source package automatically created by stdeb 0.10.0

 -- mIcHy AmRaNe <6m63er17c@mozmail.com>  Mon, 08 Jul 2024 00:28:22 +0200


okadminfinder/okadminfinder.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import argparse
import sys

import trio
from trio import TrioDeprecationWarning
import warnings

from okadminfinder import _classes


def main():
    classes = _classes.okadminfinder()
    warnings.filterwarnings(action="ignore", category=TrioDeprecationWarning)

    with classes.credit():
        parser = argparse.ArgumentParser(
            formatter_class=lambda prog: argparse.HelpFormatter(
                prog, max_help_position=30, width=90
            )
        )
        parser.add_argument(
            "-u",
            "--url",
            default=False,
            help="Target URL (e.g. 'www.example.com' or 'example.com')",
        )
        parser.add_argument(
            "-d",
            "--dns",
            action="store_true",
            default=False,
            help="Find DNS subdomains",
        )
        parser.add_argument(
            "-w",
            "--wordlist",
            default=False,
            metavar="file",
            help="Custom wordlist (e.g. /usr/share/wordlist/sub-domains.txt)",
        )
        parser.add_argument(
            "-t",
            "--tor",
            action="store_true",
            default=False,
            help="Use Tor anonymity network",
        )
        parser.add_argument(
            "-p",
            "--proxy",
            default=False,
            help="""To use an HTTP(S) proxy (e.g '127.0.0.1:8080')
                    To use an SOCKS(4/5) proxy (e.g 'socks5://127.0.0.1:8080')""",
        )
        parser.add_argument(
            "-r",
            "--random-agent",
            action="store_true",
            default=False,
            dest="rand",
            help="Use randomly selected User-Agent",
        )
        parser.add_argument(
            "-D", "--debug", action="store_true", default=False, help="Debug mode"
        )
        if len(sys.argv) <= 1:
            parser.print_usage()
            sys.exit(2)
        else:
            args = parser.parse_args()

            # user-agent
            if args.rand:
                headers = classes.get_agents()
            else:
                headers = None

            # proxy
            if args.proxy:
                if args.url is False:
                    parser.print_usage()
                    quit(0)
                elif args.tor is not False:
                    parser.print_usage()
                    print(" \n use only Tor or only Proxy not both at a time")
                    quit(0)
                else:
                    if str(args.proxy)[0:5] == "socks":
                        prox = str(args.proxy)
                        classes.proxy(prox, headers)
                        proxies = classes.proxy(prox, headers)

                    else:
                        prox = str(args.proxy)
                        prox = {
                            "http://": f"http://{prox}",
                            "https://": f"http://{prox}",
                        }
                        classes.proxy(prox, headers)
                        proxies = classes.proxy(prox, headers)

            if args.tor:
                if args.url is False:
                    parser.print_usage()
                    quit(0)
                elif args.proxy is not False:
                    parser.print_usage()
                    quit(0)
                else:
                    args.proxy = "socks5://127.0.0.1:9050"
                    prox = str(args.proxy)
                    classes.proxy(prox, headers)
                    proxies = classes.proxy(prox, headers)

            else:
                proxies = None

            # debug
            if args.debug:
                debug = args.debug
            else:
                debug = False

            # url
            if args.url:
                website = args.url
                wordlist = args.wordlist
                dns = args.dns
                trio.run(classes.url, website, headers, proxies, dns, wordlist, debug)


if __name__ == "__main__":
    main()
